src/processTree.ts

```diff
@@ -128,24 +128,31 @@ export function getProcesses(one: (pid: number, ppid: number, command: string, a
 			proc = spawn('/bin/ps', [ '-ax', '-o', 'pid,ppid,comm:20,command' ]);
 			proc.stdout?.setEncoding('utf8');
 			proc.stdout?.on('data', lines(line => {
+				const PROCESS_REGEX = /^\s*(?<pid>\d+)\s+(?<ppid>\d+)\s+(?<shortName>.+?)\s+(?<fullCommand>.+)$/ium
```

Comment:
Since a minor change in the output width can resolve your issue, can we stick to the old approach and apply the following patch? WDYT?
```javascript
            proc = spawn('/bin/ps', [ '-ax', '-o', 'pid:6,ppid:6,comm:20,command' ]);	// we specify the column width explicitly
			proc.stdout.setEncoding('utf8');
			proc.stdout.on('data', lines(line => {

				// the following substr arguments must match the column width specified for the "ps" command above
				const pid = Number(line.substring(0, 6));
				const ppid = Number(line.substr(7, 13));
				let command = line.substring(14, 34).trim();
				let args = line.substring(35);
```

Reply:
Yeah, we totally can! So, I will apply patch in a couple of hours, may be faster. Though I strongly advocate for preserving this part of patch:
```typescript
let pos = fullCommand.indexOf(shortName);
const commandEndPositionMaybe = fullCommand.indexOf(" ", pos + shortName.length)
const commandEndPosition = commandEndPositionMaybe < 0 ? fullCommand.length : commandEndPositionMaybe;

const command = fullCommand.substring(0, commandEndPosition)
const args = fullCommand.substring(commandEndPosition).trimStart()
```

In my opinion, it facilitates code readability compared to this:
```typescript
let pos = args.indexOf(command);
if (pos >= 0) {
	pos = pos + command.length;
	while (pos < args.length) {
		if (args[pos] === ' ') {
			break;
		}
		pos++;
	}
	command = args.substr(0, pos);
	args = args.substr(pos + 1);
}
```

*Note*: these comments are relevant for both versions
```
// binaries with spaces in path may not work
// possible solution to read directly from /proc
```